packages/runtime/src/plugins/demand-control/calculateCost.ts

```diff
@@ -0,0 +1,251 @@
+                              {
+                                extensions: {
+                                  code: 'COST_QUERY_PARSE_FAILURE',
+                                },
+                              },
+                            );
+                          }
+                          factorSet = true;
+                        }
+                      }
+                      if (listSizeAnnotation.sizedFields?.length) {
+                        for (const sizedField of listSizeAnnotation.sizedFields) {
+                          fieldFactorMap.set(sizedField, slicingArgumentFactor);
+                        }
+                      } else {
+                        factor = slicingArgumentFactor;
+                      }
+                    } else if ('assumedSize' in listSizeAnnotation) {
+                      const assumedSizeVal = listSizeAnnotation.assumedSize;
+                      const numValue = Number(assumedSizeVal);
+                      if (numValue && !isNaN(numValue)) {
+                        factor = numValue;
+                      }
+                    }
+                  }
+                }
+              } else if (listSize && returnType) {
+                const depth = getDepthOfListType(returnType);
+                if (depth > 0) {
+                  factor = listSize * depth;
+                }
+              }
+              factorQueue.push(factor);
+              /** Calculate factor end */
+
+              if (returnType) {
+                const namedReturnType = getNamedType(returnType);
+                if (isIntrospectionType(namedReturnType)) {
+                  return;
+                }
+                const namedReturnTypeAnnotations =
+                  getDirectiveExtensions<DemandControlDirectives>(
+                    namedReturnType,
+                    schema,
+                  );
+                if (namedReturnTypeAnnotations.cost) {
+                  for (const costAnnotation of namedReturnTypeAnnotations.cost) {
+                    if (costAnnotation?.weight) {
+                      const weight = Number(costAnnotation?.weight);
+                      if (weight && !isNaN(weight)) {
+                        currentFieldCost += weight;
+                      }
+                    }
+                  }
+                } else {
+                  currentFieldCost += typeCost(namedReturnType);
+                }
+              }
+              if (currentFieldCost) {
+                cost += timesFactor(currentFieldCost);
+              }
+            }
+          },
+          leave() {
+            factorQueue.pop();
+          },
+        },
+        Directive() {
+          const directive = typeInfo.getDirective();
+          if (directive) {
+            const directiveCostAnnotations = getDirective(
+              schema,
+              directive,
+              'cost',
```

Comment:
@ardatan it's not always `cost`. 

```
extend schema @link(url: "...", import: [{ name: "@cost", as: "@priceTag" }])
```

May end up with `@priceTag` in the supergraph.

Reply:
We have a helper util in the hive repo for verifying link versions and extracting the appropriate name.
We've been using it internally, but could publish it. https://github.com/graphql-hive/console/tree/main/packages/libraries/federation-link-utils